src/stream.c

```diff
@@ -1359,11 +1365,24 @@ void _uvc_populate_frame(uvc_stream_handle_t *strmh) {
   frame->capture_time_finished = strmh->capture_time_finished;
 
   /* copy the image data from the hold buffer to the frame (unnecessary extra buf?) */
-  if (frame->data_bytes < strmh->hold_bytes) {
-    frame->data = realloc(frame->data, strmh->hold_bytes);
-  }
-  frame->data_bytes = strmh->hold_bytes;
-  memcpy(frame->data, strmh->holdbuf, frame->data_bytes);
+  /* also: some cameras send a corrupt frame whose buffer size is smaller than what would be required to hold height*width  */
+  if (frame->data_bytes < strmh->hold_bytes ||
```

Comment:
This probably could be simplified, it seems you do the same checks inside and outside the `if`, and probably could simplify this logic by adding more cases to the outermost if.